docs/Reference/Plugin-API-Interfaces.md

```diff
@@ -37,4 +37,16 @@ The following table lists interfaces providing services you can retrieve.
 
 To use the interfaces in your plugin, ensure the
 [Gradle build file](https://github.com/PegaSysEng/PluginsAPIDemo/blob/master/build.gradle) contains
-the `https://hyperledger-org.bintray.com/besu-repo` repository and the `plugin-api` dependency.
\ No newline at end of file
+the `https://hyperledger-org.bintray.com/besu-repo` repository and the `plugin-api` dependency.
+
+!!! warning "Known issue"
+
+    As indicated in [issue #406](https://github.com/hyperledger/besu-docs/issues/406),
```

Comment:
_personally_ I think the message almost needs to be:
It is recognised that an additional step may need to exist in the plugin lifecycle, in between register and start.
The issue being that inter-module dependencies may mean that some modules need to be ready prior to start being executed.
If your plugin is required to offer service prior to "start" phase (and the plugin lifecycle hasn't be updated yet), it is recommended that the following work around be used by plugin developers:
* During register phase
** register required cmdline options with the PicoCli service (as normal)
** Register a memoised supplier of your service with the BesuContext (as you are unable to register a concrete version due to lack of cmdline args being available)
* During the start phase - take any further actions required.
The goal being, that when the core Besu engine attempts to use your plugin (which is guaranteed to occur after cmdline parsing, but may be before "start") the memoised supplier will be executed, and will thus create the concrete service _on first usage_ (rather than at start).